### PR TITLE
Add CLIP resources

### PR DESCRIPTION
**Pull request by eenzeenee**
# What does this PR do?

Adds resources of CLIP according to [this issue](https://github.com/huggingface/transformers/issues/20055.)

Part of #20055

## Before submitting
- [x ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu, @jungnerd, @wonhyeongseo may you please review this PR?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by jungnerd** [comment]
LGTM! Thanks for adding resources for CLIP. 
By the way, please fix `Part of #20555` to `Part of #20055` since 20055 is the issue of `Model resources contribution`.

**Review by wonhyeongseo — APPROVED** [review]
LGTM!

File: docs/source/en/model_doc/clip.md [hunk 1]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
 - CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
 
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing), a demo to using pretrained CLIP to inference captions with beam search. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:91]:
```suggestion
- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
```

File: docs/source/en/model_doc/clip.md [hunk 2]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+
+<PipelineTag pipeline="text-to-image"/>
+
+- A [notebook](https://colab.research.google.com/drive/1py7au62Ky4ZdLtFJP0D4shNkIVym7JS6), a demo to generating image with prompting style tokens using pretrained CLIP and vqgan. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:95]:
```suggestion
- A [notebook](https://colab.research.google.com/drive/1py7au62Ky4ZdLtFJP0D4shNkIVym7JS6) on generating images with prompting style tokens with pretrained CLIP and VQGAN. 🌎
```

File: docs/source/en/model_doc/clip.md [hunk 3]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+
+🚀 Deploy
+
+- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing), a guide on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:99]:
```suggestion
- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
```

File: docs/source/en/model_doc/clip.md [hunk 4]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb), a demo to image retrieval with how to show similarity score. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:100]:
```suggestion
- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
```

File: docs/source/en/model_doc/clip.md [hunk 5]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing), a demo showing how to map images and texts to the same vector space using Multilingual CLIP. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:101]:
```suggestion
- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
```

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution! 🤗

File: docs/source/en/model_doc/clip.md (hunk 5 above)
Comment (NielsRogge) [inline docs/source/en/model_doc/clip.md:101]: All these notebook are showcasing CLIP for inference, not for deployment. Deployment is when you export the model to ONNX for instance, optimize the model by applying quantization, etc.

**Review by NielsRogge — COMMENTED** [review]
Thanks for contributing resources! It's very helpful, however, a lot of resources here are a bit random. 

* assigning the "text-to-image" pipeline tag to CLIP is a bit weird since CLIP is only a text encoder, vision encoder model. It can't be used for generating images using text. It is however used as a building block in Stable Diffusion for instance to condition the model on text prompts.
* the "deploy" section includes several resources which are not about deployment. Deployment is about optimizing the model for inference, using tools like ONNX, 🤗 Optimum, quantization, etc.
* the "inference" section includes a notebook about... explainability. Then it makes more sense to include this in a separate "explainability" section. Inference is about showcasing predictions with the model

Some relevant resources for CLIP include:
- fine-tuning CLIP => HF provides a training script for this: https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text.
- fine-tuning CLIP on additional (image, text) pairs: https://huggingface.co/blog/fine-tune-clip-rsicd.

File: docs/source/en/model_doc/clip.md [hunk 6]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
 - CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
 
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
+
+<PipelineTag pipeline="text-to-image"/>
+
+- A [notebook](https://colab.research.google.com/drive/1py7au62Ky4ZdLtFJP0D4shNkIVym7JS6) on generating images with prompting style tokens with pretrained CLIP and VQGAN. 🌎
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:95]:
```suggestion

```

File: docs/source/en/model_doc/clip.md [hunk 7]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
 - CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
 
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
+
+<PipelineTag pipeline="text-to-image"/>
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:93]:
```suggestion
```

File: docs/source/en/model_doc/clip.md [hunk 8]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+
+- A [notebook](https://colab.research.google.com/drive/1py7au62Ky4ZdLtFJP0D4shNkIVym7JS6) on generating images with prompting style tokens with pretrained CLIP and VQGAN. 🌎
+
+🚀 Deploy
+
+- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
+- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
+- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
+
+⚡️ Inference
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:103]:
```suggestion
📓 Explainability
```

File: docs/source/en/model_doc/clip.md [hunk 9]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
 
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
+
+<PipelineTag pipeline="text-to-image"/>
+
+- A [notebook](https://colab.research.google.com/drive/1py7au62Ky4ZdLtFJP0D4shNkIVym7JS6) on generating images with prompting style tokens with pretrained CLIP and VQGAN. 🌎
+
+🚀 Deploy
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:97]:
```suggestion
**Image retrieval**
```

File: docs/source/en/model_doc/clip.md [hunk 10]
```diff
@@ -86,6 +86,24 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+
+- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
+- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
+- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
+
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:102]: A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMBD](https://www.themoviedb.org/) datasets.

File: docs/source/en/model_doc/clip.md [hunk 11]
```diff
@@ -86,6 +86,23 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
...
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
+
+
+
+**Image retrieval**
+
+- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
+- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
+- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
+A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMBD](https://www.themoviedb.org/) datasets.
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:100]:
```suggestion
A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMBD](https://www.themoviedb.org/) datasets.🌎
```

File: docs/source/en/model_doc/clip.md [hunk 12]
```diff
@@ -86,6 +86,23 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
 - CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
 
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search. 🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:91]:
```suggestion
- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search for image captioning. 🌎
```

**Review by stevhliu — COMMENTED** [review]
Thanks for fixing, can you also include the [training script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text) and [blog post](https://huggingface.co/blog/fine-tune-clip-rsicd) @NielsRogge  linked to?

**Comment by eenzeenee** [comment]
> Thanks for fixing, can you also include the [training script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text) and [blog post](https://huggingface.co/blog/fine-tune-clip-rsicd) @NielsRogge linked to?

It seems to appear in 86 and 87 lines. Do you want me to change the description to something else?

**Comment by stevhliu** [comment]
> Do you want me to change the description to something else?

Should be good then! 👍

File: docs/source/en/model_doc/clip.md [hunk 13]
```diff
@@ -86,6 +86,23 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
 - CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:87]:
```suggestion
- This [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text) shows how to train a CLIP-like vision-text dual encoder model using a pre-trained vision and text encoder using [COCO dataset](https://cocodataset.org/#home).
```

File: docs/source/en/model_doc/clip.md [hunk 14]
```diff
@@ -86,6 +86,23 @@ A list of official Hugging Face and community (indicated by 🌎) resources to h
 - A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
```
Comment (eenzeenee) [inline docs/source/en/model_doc/clip.md:86]:
```suggestion
- [Fine tuning CLIP with Remote Sensing (Satellite) images and captions](https://huggingface.co/blog/fine-tune-clip-rsicd), a blog post about how to fine-tune CLIP with [RSICD dataset](https://github.com/201528014227051/RSICD_optimal) and comparison of performance changes due to data augmentation.
```

File: docs/source/en/model_doc/clip.md [hunk 15]
```diff
@@ -83,8 +83,25 @@ This model was contributed by [valhalla](https://huggingface.co/valhalla). The o
...
-- A blog post on [How to fine-tune CLIP on 10,000 image-text pairs](https://huggingface.co/blog/fine-tune-clip-rsicd).
-- CLIP is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text).
+- [Fine tuning CLIP with Remote Sensing (Satellite) images and captions](https://huggingface.co/blog/fine-tune-clip-rsicd), a blog post about how to fine-tune CLIP with [RSICD dataset](https://github.com/201528014227051/RSICD_optimal) and comparison of performance changes due to data augmentation.
+- This [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/contrastive-image-text) shows how to train a CLIP-like vision-text dual encoder model using a pre-trained vision and text encoder using [COCO dataset](https://cocodataset.org/#home).
+
+<PipelineTag pipeline="image-to-text"/>
+
+- A [notebook](https://colab.research.google.com/drive/1tuoAC5F4sC7qid56Z0ap-stR3rwdk0ZV?usp=sharing) on how to use a pretrained CLIP for inference with beam search for image captioning. 🌎
+
+
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:93]:
```suggestion
```

File: docs/source/en/model_doc/clip.md [hunk 16]
```diff
@@ -83,8 +83,25 @@ This model was contributed by [valhalla](https://huggingface.co/valhalla). The o
...
+
+**Image retrieval**
+
+- A [notebook](https://colab.research.google.com/drive/1bLVwVKpAndpEDHqjzxVPr_9nGrSbuOQd?usp=sharing) on image retrieval using pretrained CLIP and computing MRR(Mean Reciprocal Rank) score. 🌎
+- A [notebook](https://colab.research.google.com/github/deep-diver/image_search_with_natural_language/blob/main/notebooks/Image_Search_CLIP.ipynb) on image retrieval and showing the similarity score. 🌎
+- A [notebook](https://colab.research.google.com/drive/1xO-wC_m_GNzgjIBQ4a4znvQkvDoZJvH4?usp=sharing) on how to map images and texts to the same vector space using Multilingual CLIP. 🌎 
+A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMBD](https://www.themoviedb.org/) datasets.🌎
```
Comment (stevhliu) [inline docs/source/en/model_doc/clip.md:100]:
```suggestion
- A [notebook](https://colab.research.google.com/github/vivien000/clip-demo/blob/master/clip.ipynb#scrollTo=uzdFhRGqiWkR) on how to run CLIP on semantic image search using [Unsplash](https://unsplash.com) and [TMBD](https://www.themoviedb.org/) datasets. 🌎
```

**Review by stevhliu — COMMENTED** [review]
A few more minor nits sorry! 😅

**Pull request merged**